src/components/Modals/NewCourse/NewCourseModal.vue

```diff
@@ -85,6 +90,9 @@ export default defineComponent({
       this.selectedCourse = result;
       this.getReqsRelatedToCourse(result);
     },
+    escape() {
+      this.$emit('update:modelValue', false);
+    },
```

Comment:
Discussed over slack, but we don't need any reset functions anymore. Moved the v-if to the parent of each component instead of keeping it in the TeleportModal, so that all of the code resets when closed and re-opened.